### ISSUE TITLE
App Scheme

### ISSUE THREAD

**Issue by pfrazee** [beakerbrowser/beaker#222]
Spec: https://github.com/beakerbrowser/beaker/wiki/App-Scheme

Tasks:

 - [ ] Finish spec
 - [x] App-scheme registry DB
 - [x] Registry API & permissions
 - [x] App-scheme protocol handler
 - [x] dat.json manifest update & automatic indexing
 - [x] UI for selecting applications

**Comment by pfrazee** [comment]
We're going to mix some of the ideas here with the old Web Intents spec.

(I'll write more later, but to summarize:)

We realized that the app:// scheme as we conceived it would be somewhat nightmarish to coordinate. Imagine two programs that wanted the "mail" slot. One of them supports `app://mail/compose` and the other supports `app://mail/new-message`. Users would link to both of these, and depending on which program you choose, one URL would work and the other wouldn't. That's obviously a bad experience.

What we can do, instead, is use a simpler URL scheme such as `app:share?url=github.com/pfrazee`. This simpler scheme would not have a security domain, and would not support subpaths. Instead, it *redirects* to the chosen handler.
